Standards/scs-0403-v1-csp-kaas-observability-stack.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Architecture for the Cloud Service provider Observability System for the KaaS Layer
+type: Decision Record
+status: Draft
+track: Ops
+---
+
+## Introduction
+
+Cloud Service Providers offer a variaty of products to a customer. Those can include compute resources like virtual machines, networking and identity and access management. As customers of those services build their applications upon those offered services the service provider need to ensure a certain quality level of their offerings. This is done by observing the infrastructure. Observability systems are leverage different type of telemetry data which include:
+
+1. Metrics: Usually time series data about different parameters of a system which can include e.g. CPU usage, number of active requests, health status, etc.
+2. Logs: Messages of software events during runtime
+3. Traces: More developer oriented form of logging to provide insights to an application or to analyze request flows in distributed systems.
+
+Based on those data, an alerting system can be used to to send out notifications to an Operations Team if a system behaves abnormally. Base on the telemetry data the Operations Team can find the issue, work on it and mitigate future incidents.
+
+## Motivation
+
+Currently, only the IaaS Layer of the SCS Reference Implementation has a an Observability Stack consisting of tools like Prometheus, Grafana, and Alertmanager as well as several Exporters toextract monitoring data from the several OpenStack components and additional software that is involved in the Reference Implementation. As the Kubernetes as a Service Layer becomes more and more important and the work on the Cluster API approach to create customer clusters progresses further, an observability solution for this layer is also needed. CSP should be able to watch over customer clusters and intervene if cluster get in a malfunctioning state. For this, a toolset and architecture is need which is proposed in this ADR.
+
+## Requirements
+
+A survey was conducted to gather the needs and requirements of a CSP when providing Kubernetes as a Service. The results of the Survey (Questions with answers) were the following:
```

Comment:
I understood the  Survey more as hints for the requirements definition. 
IMO the ADR requirements section should not contain the QA blocks that contain e.g. `Do you have an observabiltiy infrastructure, if yes, how it is built` 

I would like to suggest re-format this section e.g. as follows:

KaaS observability solution SHOULD gather the following metrics ... 

KaaS observability solution SHOULD define alerts based on collected metrics ...

Reply:
I rewrote the requirements section and put the survey results as reference to the end.